disttask/framework/storage/task_table.go

```diff
@@ -0,0 +1,356 @@
+	return &GlobalTaskManager{
+		ctx: ctx,
+		se:  se,
+	}
+}
+
+// NewSubTaskManager creates a new sub task manager.
+func NewSubTaskManager(ctx context.Context, se sessionctx.Context) *SubTaskManager {
+	ctx = util.WithInternalSourceType(ctx, kv.InternalDistTask)
+	return &SubTaskManager{
+		ctx: ctx,
+		se:  se,
+	}
+}
+
+// GetGlobalTaskManager gets the global task manager.
+func GetGlobalTaskManager() (*GlobalTaskManager, error) {
+	v := globalTaskManagerInstance.Load()
+	if v == nil {
+		return nil, errors.New("global task manager is not initialized")
+	}
+	return v, nil
+}
+
+// SetGlobalTaskManager sets the global task manager.
+func SetGlobalTaskManager(is *GlobalTaskManager) {
+	globalTaskManagerInstance.Store(is)
+}
+
+// GetSubTaskManager gets the sub task manager.
+func GetSubTaskManager() (*SubTaskManager, error) {
+	v := subTaskManagerInstance.Load()
+	if v == nil {
+		return nil, errors.New("subTask manager is not initialized")
+	}
+	return v, nil
+}
+
+// SetSubTaskManager sets the sub task manager.
+func SetSubTaskManager(is *SubTaskManager) {
+	subTaskManagerInstance.Store(is)
+}
+
+// execSQL executes the sql and returns the result.
+// TODO: consider retry.
+func execSQL(ctx context.Context, se sessionctx.Context, sql string, args ...interface{}) ([]chunk.Row, error) {
+	rs, err := se.(sqlexec.SQLExecutor).ExecuteInternal(ctx, sql, args...)
+	if err != nil {
+		return nil, err
+	}
+	if rs != nil {
+		rows, err := sqlexec.DrainRecordSet(ctx, rs, 1)
+		if err != nil {
+			return nil, err
+		}
+		err = rs.Close()
+		if err != nil {
+			return nil, err
+		}
+		return rows, err
+	}
+	return nil, nil
+}
+
+// row2GlobeTask converts a row to a global task.
+func row2GlobeTask(r chunk.Row) *proto.Task {
+	task := &proto.Task{
+		ID:           r.GetInt64(0),
+		Type:         r.GetString(1),
+		DispatcherID: r.GetString(2),
+		State:        r.GetString(3),
+		Meta:         r.GetBytes(6),
+		Concurrency:  uint64(r.GetInt64(7)),
+		Step:         r.GetInt64(8),
+	}
+	// TODO: convert to local time.
+	task.StartTime, _ = r.GetTime(4).GoTime(time.UTC)
+	task.StateUpdateTime, _ = r.GetTime(5).GoTime(time.UTC)
+	return task
+}
+
+// AddNewTask adds a new task to global task table.
+func (stm *GlobalTaskManager) AddNewTask(tp string, concurrency int, meta []byte) (int64, error) {
+	stm.mu.Lock()
+	defer stm.mu.Unlock()
+
+	_, err := execSQL(stm.ctx, stm.se, "insert into mysql.tidb_global_task(type, state, concurrency, meta, state_update_time) values (%?, %?, %?, %?, %?)", tp, proto.TaskStatePending, concurrency, meta, time.Now().UTC().String())
+	if err != nil {
+		return 0, err
+	}
+
+	rs, err := execSQL(stm.ctx, stm.se, "select @@last_insert_id")
+	if err != nil {
+		return 0, err
+	}
+
+	return strconv.ParseInt(rs[0].GetString(0), 10, 64)
+}
+
+// GetNewTask get a new task from global task table, it's used by dispatcher only.
+func (stm *GlobalTaskManager) GetNewTask() (task *proto.Task, err error) {
```

Comment:
Is it better to provide a function to get unfinished tasks?  Otherwise, when TiDB downs, the task being processed will not get.

Reply:
It can be added later when it's needed.